packages/nextjs/src/server/clerkMiddleware.ts

```diff
@@ -76,6 +76,11 @@ export const clerkMiddleware: ClerkMiddleware = withLogger('clerkMiddleware', lo
     logger.enable();
   }
 
+  const hasProvidedRuntimeOptions = params.secretKey || params.signInUrl || params.signInUrl;
+  if (hasProvidedRuntimeOptions) {
+    assertKey(ENCRYPTION_KEY, () => errorThrower.throwMissingEncryptionKeyError());
+  }
```

Comment:
Maybe it's not necessary, but we had talked about defaulting to using the secret key as the encryption key **if** no secret key is provided as an option to the middleware. This prevents everyone from needing to provide an encryption key.

So I think we should adjust the check to only error if secret key is provided and no encryption key is found

Reply:
+1, considering the secret key is the only bit from the sensitive payload and therefore requires encryption.

We need to make sure that the SDK will throw in an error if there's not a `CLERK_SECRET_KEY` defined as well.